conbench/app/_plots.py

```diff
@@ -168,13 +169,21 @@ def _should_format(data, unit):
     return should, axis_unit
 
 
+def _insert_nans(some_list: list, indexes: List[int]):
+    """Insert np.NaNs into a list before the given indexes."""
+    for ix in sorted(indexes, reverse=True):
+        some_list.insert(ix, np.NaN)
```

Comment:
Interesting! Maybe at some point these ideas might become relevant:
- https://wiki.python.org/moin/TimeComplexity
- https://pypi.org/project/blist/

Reply:
Haha! I did think about performance here... but if we have enough data points in these lists that inserting a few values causes any noticeable performance degradation, I think rendering the plot itself is going to really blow up performance first 🙂